contentcuration/contentcuration/management/commands/set_file_duration.py

```diff
@@ -30,7 +34,42 @@ def extract_duration_of_media(f_in):
         ],
         stdin=f_in,
     )
-    return int(float(result.decode("utf-8").strip()))
+    result = result.decode("utf-8").strip()
+    try:
+        # return int(float(result))
+        raise ValueError
+    except ValueError:
+        # This can happen if ffprobe returns N/A for the duration
+        # So instead we try to stream the entire file to get the value
+        f_in.seek(0)
+        result = subprocess.run(
+            [
+                "ffmpeg",
+                "-i",
+                "pipe:",
+                "-f",
+                "null",
+                "-",
+            ],
+            stdin=f_in,
+            stderr=subprocess.PIPE
+        )
```

Comment:
That was my inference from the error initially, but I am not sure it was correct. Searches for why the duration might turn up `N/A` pointed to looking at the stream metadata rather than the container, but looking at my code again showed that I was already using the container inspection method.